Description: Generate service file with correct path to daemon
 The daemon might be installed in a different location than the default,
 in which case the service file will not work.
 Generate the unit at build time with the path specified by the user with
 make install.
Author: Luca Boccassi <bluca@debian.org>
Forwarded: https://github.com/smuellerDD/jitterentropy-rngd/pull/3
--- a/Makefile
+++ b/Makefile
@@ -39,9 +39,11 @@ install: strip
 	$(INSTALL) -D -m 0755 -s $(NAME) $(DESTDIR)$(PREFIX)/sbin/$(NAME)
 	$(INSTALL) -D -m 0644 $(NAME).1 $(DESTDIR)$(PREFIX)/share/man/man1/$(NAME).1
 	gzip -9 $(DESTDIR)$(PREFIX)/share/man/man1/$(NAME).1
+	sed "s|@PATH@|$(PREFIX)/sbin|" jitterentropy.service.in > jitterentropy.service
 
 clean:
 	@- $(RM) $(NAME)
 	@- $(RM) $(OBJS)
+	@- $(RM) jitterentropy.service
 
 distclean: clean
--- a/jitterentropy.service
+++ /dev/null
@@ -1,24 +0,0 @@
-# Systemd configuration file
-#
-# The Jitterentropy RNGd does not depend on any other system
-# services. Futhermore, we want to start it as early as possible
-# in the boot cycle so that other services requiring
-# random numbers (like SSHD, TLS-based services) benefit
-# from a properly seeded /dev/?random
-
-[Unit]
-Description=Jitterentropy Gatherer Daemon
-DefaultDependencies=no
-After=local-fs.target
-Before=sysinit.target
-
-[Service]
-ExecStart=/usr/local/sbin/jitterentropy-rngd
-#PrivateTmp=yes
-#PrivateDevices=yes
-#ReadOnlyDirectories=/
-#ReadWriteDirectories=/dev
-
-[Install]
-WantedBy=basic.target
-Alias=jitterentropy-rngd.service
--- /dev/null
+++ b/jitterentropy.service.in
@@ -0,0 +1,24 @@
+# Systemd configuration file
+#
+# The Jitterentropy RNGd does not depend on any other system
+# services. Futhermore, we want to start it as early as possible
+# in the boot cycle so that other services requiring
+# random numbers (like SSHD, TLS-based services) benefit
+# from a properly seeded /dev/?random
+
+[Unit]
+Description=Jitterentropy Gatherer Daemon
+DefaultDependencies=no
+After=local-fs.target
+Before=sysinit.target
+
+[Service]
+ExecStart=@PATH@/jitterentropy-rngd
+#PrivateTmp=yes
+#PrivateDevices=yes
+#ReadOnlyDirectories=/
+#ReadWriteDirectories=/dev
+
+[Install]
+WantedBy=basic.target
+Alias=jitterentropy-rngd.service
